tracer/test/Datadog.Trace.Security.IntegrationTests/IAST/Deduplicaation/DeduplicationTests.cs

```diff
@@ -0,0 +1,54 @@
+// <copyright file="DeduplicationTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NETCOREAPP
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Datadog.Trace.TestHelpers;
+using VerifyXunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.Security.IntegrationTests.Iast;
+
+[UsesVerify]
+public class DeduplicationTests : TestHelper
+{
+    private const string ExpectedOperationName = "weak_hashing";
+    private static readonly Regex LocationMsgRegex = new(@"(\S)*""location"": {(\r|\n){1,2}(.*(\r|\n){1,2}){0,3}(\s)*},");
+
+    public DeduplicationTests(ITestOutputHelper output)
+        : base("Deduplication", output)
+    {
+        SetServiceVersion("1.0.0");
+    }
+
+    [SkippableTheory]
+    [Trait("Category", "EndToEnd")]
+    [Trait("RunOnWindows", "True")]
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task SubmitsTraces(bool deduplicationEnabled)
```

Comment:
nit: I would personally prefer to re-use the existing weak cipher etc tests for this, rather than adding an extra sample. Every extra sample we add slows down the _build_ stage, which slows down every other stage in CI. Not required, but would be preferable IMO 🙂

Reply:
Yes, I thought about it. The problem is that for deduplication, some .net versions are better than others when providing the file and line in the stack, and some provide the method name instead the file (as expected, the newest are better). This means that the behaviour of the deduplication in the already existing tests varies slightly depending on the framework because the information that it receives is slightly different. Reusing the already existing tests would require to cover a few different cases that would increase the complexity, so, finally, I decided to go for a new, very simple, test app.